src/app/page/UserProfile/UserOverview.jsx

```diff
@@ -1,7 +1,11 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { H5 } from "../../component";
 import { Grid, Avatar, TextField } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
+import addressLookUp from "../../utils/addressLookUp";
+import { useSelector } from "react-redux";
+import { withRouter } from "react-router-dom";
+import { withFirestore } from "react-redux-firebase";
```

Comment:
I prefer useFirestore() instead of withFirestore. It means 1 less prop PLUS will make testing a LOT easier.

Reply:
I have another suggestion, using firestoreConnect and get your props like this;

const mapStateToProps = (state, ownProps) => {
  const missionId = ownProps.match.params.id;
  const mission = state.firestore.data.missions && state.firestore.data.missions[missionId];
  return {
    auth: state.firebase.auth,
    missionId,
    mission,
  };
};
export default compose(
  connect(mapStateToProps),
  firestoreConnect((props) => {
    const missionId = props.missionId;
    if (!props.auth.uid) return [];
    return [
      {
        collection: "missions",
        doc: missionId,
      },
    ];
  })
)(MissionDetailsPage);

it allows more consistency for how you getting your stuff, I find useAnything... a bit too magical and hard to find